### ISSUE TITLE
Bug: Support typescript version 5.6.2

### ISSUE THREAD

**Issue by samuelneff** [typescript-eslint/typescript-eslint#10066]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### Repro Code

```TypeScript
npm i -D typescript^5.6.2
eslint src
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Eslint runs without warning.

Is there a reason it can't support `5.6.*` to reduce maintenance burden?

### Actual Result

Warning displayed

```text
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=4.7.4 <5.6.0

YOUR TYPESCRIPT VERSION: 5.6.2

Please only submit bug reports when using the officially supported version.

=============
```

### Additional Info

There was a PR #9997 to update support but it was closed without merging and without discussion.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.7.0` |
| `@typescript-eslint/parser`            | `8.7.0` |
| `@typescript-eslint/typescript-estree` | `8.7.0` |
| `@typescript-eslint/type-utils`        | `8.7.0` |
| `@typescript-eslint/utils`             | `8.7.0` |
| `TypeScript`                           | `5.6.2` |
| `ESLint`                               | `9.11.1` |
| `node`                                 | `20.13.1` |

**Comment by auvred** [comment]
Please see the pinned issue https://github.com/typescript-eslint/typescript-eslint/issues/9653
